force-app/main/default/classes/DMLOperations.cls

```diff
@@ -97,10 +111,14 @@ public with sharing class DMLOperations {
     public static void updateOpportunityStage(Id oppId, String newStage) {
         // Retrieve the Opportunity from the database
         // [SELECT Id FROM Opportunity WHERE Id = :oppId]
+        Opportunity updatedOpportunity = [SELECT Id FROM Opportunity WHERE Id = :oppId];
 
         // Update the Opportunity's stage
+        updatedOpportunity.StageName = newStage;
 
         // Update the Opportunity in the database
+        update updatedOpportunity;
+
```

Comment:
great job. Do you think you could solve this with out the query?

Reply:
How?

Reply:
Opportunity updatedOpportunity = new Opportunity(Id = oppId);